### DISCUSSION TITLE
ColumnAttribute.StringLength 建议

### DISCUSSION THREAD

**Discussion by shaukei** [dotnetcore/FreeSql#1370] [General]
`[JsonProperty, Column(DbType = "varchar(10)")]
		public string item_code { get; set; }`

希望设置号DbType（有长度信息）以后
可以有选项设置数据超过时候的处理方式：
1、截断后插入
2、抛异常

**Comment by 2881099** [comment]
最开始是可以截断的，后面他们老报错就取消强制截断了。

异常，可以使用 fsql.Aop.ConfigEntityProperty 事件处理。

```c#
fsql.Aop.ConfigEntityProperty += (_, e) =>
{
    if (e.Property.GetCustomAttribute<ColumnAttribute>()?.DbType......)
        throw new Exception...
};
```

**Reply by shaukei** [reply]
截断插入呢？有方法不？

**Comment by 2881099** [comment]
fsql.Aop.AuditValue 事件看一下文档，可以判断长度截断

**Reply by shaukei** [reply]
谢谢！

`if (e.Column.CsType == typeof(string) && e.Column.DbSize < e.Value?.ToString().Length)
                {
                    e.Value = e.Value.ToString().Substring(0, e.Column.DbSize);
                }`
